CMakeLists.txt

```diff
@@ -37,7 +37,8 @@ set(${PROJECT_NAME}_SRCS
   src/meta_object.cpp
   src/multi_library_class_loader.cpp
 )
-add_library(${PROJECT_NAME} ${${PROJECT_NAME}_SRCS})
+file(GLOB ${PROJECT_NAME}_HDRS "include/${PROJECT_NAME}/*.h") # allow headers to be shown in CodeBlocks/QtCreator
```

Comment:
Globbing source files in CMake is discouraged: https://cmake.org/cmake/help/v3.5/command/file.html?highlight=glob#file

> We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate.